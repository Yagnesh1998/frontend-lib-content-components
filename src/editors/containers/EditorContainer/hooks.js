import { useSelector } from 'react-redux';

import { RequestKeys } from '../../data/constants/requests';
import { selectors } from '../../data/redux';
import * as appHooks from '../../hooks';
import analyticsEvt from '../../data/constants/analyticsEvt';

export const {
  navigateCallback,
  nullMethod,
  saveBlock,
} = appHooks;

export const setAssetToStaticUrl = (images, getContent) => {
  let content = getContent();
  const imageUrls = [];
  const imgsArray = Object.values(images);
  imgsArray.forEach(image => {
    imageUrls.push({ portableUrl: image.portableUrl, displayName: image.displayName });
  });
  const imageSrcs = content.split('src="');
  imageSrcs.forEach(src => {
    if (src.startsWith('/asset') && imageUrls.length > 0) {
      const nameFromEditorSrc = src.substring(src.lastIndexOf('@') + 1, src.indexOf('"'));
      const nameFromStudioSrc = nameFromEditorSrc.substring(nameFromEditorSrc.indexOf('/') + 1);
      let portableUrl;
      imageUrls.forEach((url) => {
        if (url.displayName === nameFromEditorSrc || url.displayName === nameFromStudioSrc) {
          portableUrl = url.portableUrl;
        }
      });
      if (portableUrl) {
        const currentSrc = src.substring(0, src.indexOf('"'));
        const updatedContent = content.replace(currentSrc, portableUrl);
        content = updatedContent;
      }
    }
  });
  return content;
};

export const handleSaveClicked = ({ getContent, dispatch }) => {
  const destination = useSelector(selectors.app.returnUrl);
  const analytics = useSelector(selectors.app.analytics);
  const images = useSelector(selectors.app.images);
  return () => saveBlock({
    content: setAssetToStaticUrl(images, getContent),
    destination,
    analytics,
    dispatch,
  });
};
export const handleCancelClicked = ({ onClose }) => {
  if (onClose) {
    return onClose;
  }
  return navigateCallback({
    destination: useSelector(selectors.app.returnUrl),
    analyticsEvent: analyticsEvt.editorCancelClick,
    analytics: useSelector(selectors.app.analytics),
  });
};
export const isInitialized = () => useSelector(selectors.app.isInitialized);
export const saveFailed = () => useSelector((state) => (
  selectors.requests.isFailed(state, { requestKey: RequestKeys.saveBlock })
));
