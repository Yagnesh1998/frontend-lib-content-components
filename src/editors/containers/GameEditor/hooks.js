import { useState, useEffect } from 'react';
import { Form } from '@edx/paragon';
import './index.scss';
