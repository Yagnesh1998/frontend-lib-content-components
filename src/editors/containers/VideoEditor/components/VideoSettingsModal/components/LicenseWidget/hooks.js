import { actions } from '../../../../../../data/redux';
import { LicenseLevel, LicenseTypes } from '../../../../../../data/constants/licenses';

export const determineLicense = ({
  isLibrary,
  licenseType,
  licenseDetails,
  courseLicenseType,
  courseLicenseDetails,
}) => {
  let level = LicenseLevel.course;
  if (licenseType) {
    if (isLibrary) {
      level = LicenseLevel.library;
    }
    else {
      level = LicenseLevel.block;
    }
  }

  return {
    license: licenseType ? licenseType : courseLicenseType,
    details: licenseType ? licenseDetails : courseLicenseDetails,
    level,
  };
};

export const determineText = ({ level }) => {
  // TODO intl
  let levelDescription = '';
  let licenseDescription = '';
  switch (level) {
    case LicenseLevel.course:
      levelDescription = "This license currently set at the course level";
      licenseDescription = "Licenses set at the course level appear at the bottom of courseware pages within your course.";
      break;
    case LicenseLevel.library:
      levelDescription = "This license currently set at the library level";
      licenseDescription = "Licenses set at the library level appear at the specific library video.";
      break;
    default: // default to block
      levelDescription = "This license is set specifically for this video";
      licenseDescription = "When a video has a different license than the course as a whole, learners see the license at the bottom right of the video player.";
      break;
  }

  return {
    levelDescription,
    licenseDescription,
  };
};

export const onSelectLicense = ({
  dispatch,
}) => (license) => {
  switch (license) {
    case LicenseTypes.allRightsReserved:
      dispatch(actions.video.updateField({ licenseType: LicenseTypes.allRightsReserved}));
      break;
    case LicenseTypes.creativeCommons:
      dispatch(actions.video.updateField({
        licenseType: LicenseTypes.creativeCommons,
        licenseDetails: {
          attribution: true,
          noncommercial: true,
          noDerivatives: true,
          shareAlike: false,
        },
      }));
      break;
  }
};

export default {
  determineLicense,
  determineText,
  onSelectLicense,
};