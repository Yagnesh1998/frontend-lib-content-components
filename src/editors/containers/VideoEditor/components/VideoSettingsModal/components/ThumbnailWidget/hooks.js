import React from 'react';
import { useDispatch } from 'react-redux';

import { actions } from '../../../../../../data/redux';

// export const parseImage = ({ reader, file }) => {
//   reader.readAsDataURL(file);
//   return
// }

export const fileInput = (imgRef) => {
  const dispatch = useDispatch()
  const ref = React.useRef();
  const click = () => ref.current.click();
  const addFile = (e) => {
    const reader = new FileReader()
    reader.onload = () => {
      imgRef.current.src = reader.result
    }
    const file = e.target.files[0];
    if (file) {
      reader.readAsDataURL(file)
      console.log(reader);
      dispatch(actions.video.updateField({ thumbnail: file }))
      console.log(imgRef);
    }
  };
  return {
    click,
    addFile,
    ref,
  };
};

export default {
  fileInput,
};
