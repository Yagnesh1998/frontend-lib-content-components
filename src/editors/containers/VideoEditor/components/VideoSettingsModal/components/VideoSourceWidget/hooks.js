import { actions } from '../../../../../../data/redux';

/**
 * deleteFallbackVideo({ fallbackVideos, dispatch })(videoUrl)
 * deleteFallbackVideo takes the current array of fallback videos, string of
 * deleted video URL and dispatch  method, and updates the redux value for
 * fallbackVideos.
 * @param {array} fallbackVideos - array of current fallback videos
 * @param {func} dispatch - redux dispatch method
 * @param {string} videoUrl - string of the video URL for the fallabck video that needs to be deleted
 */
export const deleteFallbackVideo = ({ fallbackVideos, dispatch }) => (videoUrl) => {
  const updatedFallbackVideos = [];
  let firstOccurence = true;
  fallbackVideos.forEach(item => {
    if (item === videoUrl && firstOccurence) {
      firstOccurence = false;
    } else {
      updatedFallbackVideos.push(item);
    }
  });
  dispatch(actions.video.updateField({ fallbackVideos: updatedFallbackVideos }));
};

export default { deleteFallbackVideo };
