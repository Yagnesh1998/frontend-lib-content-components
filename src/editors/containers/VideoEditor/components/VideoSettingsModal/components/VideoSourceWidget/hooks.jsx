import { actions } from '../../../../../../data/redux';

export const updateVideoURL = ({ dispatch }) => ({ e }) => dispatch(
  actions.video.updateField({
    videoSource: e.target.value
  })
);

export const updateVideoId = ({ dispatch }) => ({ e }) => dispatch(
  actions.video.updateField({
    videoId: e.target.value
  })
);

export const addFallbackVideo = ({ fallbackVideos, dispatch }) => () => dispatch(
  actions.video.updateField({ fallbackVideos: [...fallbackVideos.formValue, ''] })
);

/**
 * deleteFallbackVideo({ fallbackVideos, dispatch })(videoUrl)
 * deleteFallbackVideo takes the current array of fallback videos, string of
 * deleted video URL and dispatch  method, and updates the redux value for
 * fallbackVideos.
 * @param {array} fallbackVideos - array of current fallback videos
 * @param {func} dispatch - redux dispatch method
 * @param {string} videoUrl - string of the video URL for the fallabck video that needs to be deleted
 */
export const deleteFallbackVideo = ({ fallbackVideos, dispatch }) => (videoUrl) => {
  const updatedFallbackVideos = [];
  let firstOccurence = true;
  fallbackVideos.forEach(item => {
    if (item === videoUrl && firstOccurence) {
      firstOccurence = false;
    } else {
      updatedFallbackVideos.push(item);
    }
  });
  dispatch(actions.video.updateField({ fallbackVideos: updatedFallbackVideos }));
};

export default {
  updateVideoId,
  updateVideoURL,
  addFallbackVideo,
  deleteFallbackVideo,
};
