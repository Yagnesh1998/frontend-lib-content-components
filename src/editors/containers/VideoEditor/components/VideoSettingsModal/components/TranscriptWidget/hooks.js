import React from 'react';
import { thunkActions, actions } from '../../../../../../data/redux';
import * as module from './hooks';

export const state = {
  inDeleteConfirmation: (args) => React.useState(args),
};

export const transcriptLanguages = (transcripts) => {
  const languages = [];
  if (Object.keys(transcripts).length > 0) {
    Object.keys(transcripts).forEach(transcript => {
      languages.push(transcript);
    });
    return languages.join(', ');
  }
  return 'None';
};

export const hasTranscripts = (transcripts) => {
  if (transcripts && Object.keys(transcripts).length > 0) {
    return true;
  }
  return false;
};

export const onSelectLanguage = ({
  filename, dispatch, transcripts, languageBeforeChange,
}) => (e) => {
  const { [languageBeforeChange]: removedProperty, ...trimmedTranscripts } = transcripts;
  const newTranscripts = { [e.target.value]: { filename }, ...trimmedTranscripts };
  dispatch(actions.video.updateField({ transcripts: newTranscripts }));
};

export const replaceFileCallback = ({ language, dispatch }) => (e) => {
  dispatch(thunkActions.video.replaceTranscript({
    newFile: e.target.files[0],
    newFilename: e.target.files[0].name,
    language,
  }));
};

export const addFileCallback = ({ dispatch }) => (e) => {
  dispatch(thunkActions.video.uploadTranscript({
    file: e.target.files[0],
    filename: e.target.files[0].name,
    language: null,
  }));
};

export const fileInput = ({ onAddFile }) => {
  const ref = React.useRef();
  const click = () => ref.current.click();
  const addFile = (e) => {
    onAddFile(e);
  };
  return {
    click,
    addFile,
    ref,
  };
};

export const setUpDeleteConfirmation = () => {
  const [inDeleteConfirmation, setInDeleteConfirmation] = module.state.inDeleteConfirmation(false);
  return {
    inDeleteConfirmation,
    launchDeleteConfirmation: () => setInDeleteConfirmation(true),
    cancelDelete: () => setInDeleteConfirmation(false),
  };
};

export default {
  transcriptLanguages, fileInput, onSelectLanguage, replaceFileCallback, addFileCallback, setUpDeleteConfirmation,
};
