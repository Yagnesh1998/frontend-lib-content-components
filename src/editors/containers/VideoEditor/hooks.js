import { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { thunkActions, selectors } from '../../data/redux';
import { StrictDict } from '../../utils';
import * as module from './hooks';

export const state = StrictDict({
  durationErrors: (val) => useState(val),
  handoutErrors: (val) => useState(val),
  licenseErrors: (val) => useState(val),
  thumbnailErrors: (val) => useState(val),
  transcriptsErrors: (val) => useState(val),
  videoSourceErrors: (val) => useState(val),
});

export const errorsHook = () => {
  const [durationErrors, setDurationErrors] = module.state.durationErrors({});
  const [handoutErrors, setHandoutErrors] = module.state.handoutErrors({});
  const [licenseErrors, setLicenseErrors] = module.state.licenseErrors({});
  const [thumbnailErrors, setThumbnailErrors] = module.state.thumbnailErrors({});
  const [transcriptsErrors, setTranscriptsErrors] = module.state.transcriptsErrors({});
  const [videoSourceErrors, setVideoSourceErrors] = module.state.videoSourceErrors({});

  return {
    error: {
      duration: durationErrors,
      handout: handoutErrors,
      license: licenseErrors,
      thumbnail: thumbnailErrors,
      transcripts: transcriptsErrors,
      videoSource: videoSourceErrors,
    },
    validateEntry: () => {
      let validated = true;
      if (!module.validateDuration({ setDurationErrors })) { validated = false; }
      if (!module.validateHandout({ setHandoutErrors })) { validated = false; }
      if (!module.validateLicense({ setLicenseErrors })) { validated = false; }
      if (!module.validateThumbnail({ setThumbnailErrors })) { validated = false; }
      if (!module.validateTranscripts({ setTranscriptsErrors })) { validated = false; }
      if (!module.validateVideoSource({ setVideoSourceErrors })) { validated = false; }
      return validated;
    },
  };
};

export const validateDuration = ({ setDurationErrors }) => {
  setDurationErrors({
    fieldName: 'sample error message',
  });
  return false;
};
export const validateHandout = ({ setHandoutErrors }) => {
  setHandoutErrors({
    fieldName: 'sample error message',
  });
  return false;
};
export const validateLicense = ({ setLicenseErrors }) => {
  setLicenseErrors({
    fieldName: 'sample error message',
  });
  return false;
};
export const validateThumbnail = ({ setThumbnailErrors }) => {
  setThumbnailErrors({
    fieldName: 'sample error message',
  });
  return false;
};
export const validateTranscripts = ({ setTranscriptsErrors }) => {
  setTranscriptsErrors({
    fieldName: 'sample error message',
  });
  return false;
};
export const validateVideoSource = ({ setVideoSourceErrors }) => {
  setVideoSourceErrors({
    fieldName: 'sample error message',
  });
  return false;
};

export const getContent = () => {
  const dispatch = useDispatch();
  const currentThumbnail = useSelector(selectors.video.thumbnail);
  const originalThumbnail = useSelector(selectors.video.originalThumbnail);
  if (currentThumbnail !== originalThumbnail) {
    dispatch(thunkActions.video.uploadThumbnail({
      thumbnail: currentThumbnail,
    }));
  }
  const videoSettings = useSelector(selectors.video.videoSettings);
  return videoSettings;
};
