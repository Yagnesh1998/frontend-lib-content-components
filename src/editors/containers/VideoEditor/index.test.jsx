import React from 'react';
import { shallow } from 'enzyme';

import { errorsHook } from './hooks';
import { VideoEditor, mapDispatchToProps } from '.';

jest.mock('../EditorContainer', () => 'EditorContainer');
jest.mock('./components/VideoEditorModal', () => 'VideoEditorModal');

jest.mock('./hooks', () => ({
  errorsHook: jest.fn(() => ({
    error: 'hooks.errorsHook.error',
    validateEntry: jest.fn().mockName('validateEntry'),
  })),
  fetchVideoContent: jest.fn().mockName('fetchVideoContent'),
}));

jest.mock('../../data/redux', () => ({
  selectors: {
    video: {
      videoSettings: state => ({ videoSettings: { state } }),
    },
  },
}));

describe('VideoEditor', () => {
  const props = {
    onClose: jest.fn().mockName('props.onClose'),
  };
  errorsHook.mockReturnValue({
    error: 'errORsHooKErroR',
    validateEntry: jest.fn().mockName('validateEntry'),
  });
  describe('snapshots', () => {
    test('renders as expected with default behavior', () => {
      expect(shallow(<VideoEditor {...props} />)).toMatchSnapshot();
    });
  });
  describe('mapDispatchToProps', () => {
    test('is empty', () => {
      expect(mapDispatchToProps).toEqual({});
    });
  });
});
