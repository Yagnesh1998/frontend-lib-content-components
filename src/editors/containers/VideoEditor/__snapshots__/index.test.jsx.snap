// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoEditor snapshots renders as expected with default behavior 1`] = `
<Component
  value="hooks.errorsHook.error"
>
  <EditorContainer
    onClose={[MockFunction props.onClose]}
    validateEntry={[MockFunction validateEntry]}
  >
    <Spinner
      animation="border"
      className="m-3"
      screenreadertext="loading"
    />
  </EditorContainer>
</Component>
`;

exports[`VideoEditor snapshots renders as expected with default behavior 2`] = `
<Component
  value="hooks.errorsHook.error"
>
  <EditorContainer
    onClose={[MockFunction props.onClose]}
    validateEntry={[MockFunction validateEntry]}
  >
    <div
      className="video-editor"
    >
      <VideoEditorModal />
    </div>
  </EditorContainer>
</Component>
`;
